### ISSUE TITLE
Accessing cookie in route index action?

### ISSUE THREAD

**Issue by SmboBeast** [kriasoft/react-starter-kit#1383]
Hi guys,

I would like to authenticate routes by seeing if the user has an auth token located in the cookie.  However, I can't seem to to be able to access the document or cookies for that matter from the route indexes.  Is this the right way to do this?  How else do people authenticate routes?

Do I have to store things in redux instead?  Would that be better?

Thanks

**Comment by ucay** [comment]
By using redux, you can authenticate route like this:
```javascript
async action({ store }) {
    // Redirect to login page if user is not admin
    if (!store.getState().isAdmin) {
      return { redirect: '/login' };
    }
    return {
      title,
      component: <YourComponent />,
    };
  },
```

Read more in here #870

**Comment by SmboBeast** [comment]
@ucay Ahh, yes.  Thank you!  Glad I asked.

**Comment by SmboBeast** [comment]
@ucay 

The only thing is with using redux is that a user can't stay logged in if they move away from the site and come back.  So I would need a hybrid approach of cookies plus redux I believe.

That way, if a user comes back to the website, the cookie can tell the server they are still to remain logged in, and from there, state can be written to and state can be checked from the routes code.

Thoughts?

**Comment by pfftdammitchris** [comment]
You can use the native local storage API to store the user's cookie on their local system, or however else you want to save history. Then code some auto-loading local storage mechanism on page visit to

I use [store.js](https://github.com/marcuswestin/store.js/), it has a very simple api and quick and easy to install to your app.  The library uses the native local storage api built into javascript. Read [this](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) for more info on local storage. 

Currently I use it to persist my post submissions. (ex: when submitting a new blog post, I use it to save the "draft" so that I can come back to it later even after a computer restart)

This doesn't take into account a user-authenticated conditioning rendering where it isn't local system based but i'm sure you can combine local storage into that as well.

**Comment by SmboBeast** [comment]
Hi @pfftdammitchris,

This is the approach I used originally, however, the drawback I encountered was when someone closes the browser and visits my webpage again for the first time, it's a server render.  Because it's a server render, it has no access to localStorage, so it assumes the user has to log in again.

I had to end up using cookies, so that the server.js file can see if the user is still authenticated or not, since it has access to cookies from there.  Then, I drop all that info I need into state using Redux, and my routes can just check the store to see if they are authenticated or not.

Basically, the use of cookies was only for the first time server rendering.

Please let me know if I'm misunderstanding you, and you are suggesting a better way to do it.

**Comment by langpavel** [comment]
@SmboBeast You are on better correction.
